### PR TITLE
Add Registration form

### PR DESCRIPTION
**Pull request by pavan-testpress**
### Changes done
- Added Registration form.
- Added Email verfication.
- Added Signup View.
- Written TestCases.

### Reason for the changes
- To Register new user and verify their email address.

### Stats
- Number of Test Cases - 4 
- Number of SQL Queries - 4

### Guidelines
- [x] PEP8(Python)
- [x] Is each logical block separated with a line break?
- [x] Have you self reviewed this PR in context to the previous PR feedbacks?

### CODE REVIEW

File: onboarding/apps/accounts/views.py [hunk 1]
```diff
@@ -1,7 +1,59 @@
...
+        return HttpResponse('<span>Thank you for your email confirmation. Now you can <a href="/login/">login</a> your account.</span>')
+    else:
+        return HttpResponse('Activation link is invalid!')
+
+
+def signup(request):
+    if request.user.is_authenticated:
+        return HttpResponseRedirect(reverse('accounts:home'))
+    else:
+        if request.method == 'POST':
```
Comment (bharathwaaj) [inline onboarding/apps/accounts/views.py]: A better way to write would be
```
form = UserCreationForm(data = request.POST or None)
if form.is_valid():
    # Write code for POST operation
    return HttpResponseRedirect(success_page)

# Write code for GET operation
return render(request...)
```

File: onboarding/apps/accounts/views.py [hunk 2]
```diff
@@ -1,7 +1,59 @@
...
+                message = render_to_string('registration/account_activate_email.html', {
+                    'user': user,
+                    'domain': current_site.domain,
+                    'uid': urlsafe_base64_encode(force_bytes(user.pk)).decode(),
+                    'token': account_activation_token.make_token(user),
+                })
+                mail_subject = 'Activate your account.'
+                to_email = form.cleaned_data.get('email')
+                send_mail(mail_subject, message, 'pavan1995143.pavan@gmail.com', [to_email, ])
+                return HttpResponse('Please confirm your email address to complete the registration')
```
Comment (bharathwaaj) [inline onboarding/apps/accounts/views.py]: A POST operation should always be followed by a redirect.

Comment (bharathwaaj) [reply onboarding/apps/accounts/views.py]: https://en.m.wikipedia.org/wiki/Post/Redirect/Get

File: onboarding/tests/test_accounts/test_signup.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        response = self.client.get(reverse('accounts:signup'))
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, 'registration/signup.html')
+
+    def test_post_signup_page(self):
+        """
+        Test to create user through signup page.
+        """
+        response = self.client.post(reverse('accounts:signup'), self.user)
+        self.assertEqual(response.context['domain'], 'testserver')
```
Comment (bharathwaaj) [inline onboarding/tests/test_accounts/test_signup.py]: This test might change if sign up redirects after POST.

File: onboarding/tests/test_accounts/test_signup.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+    def test_try_to_login_before_verification(self):
+        self.client.post(reverse('accounts:signup'), self.user)
+        response = self.client.post(reverse('accounts:login'), {'username': 'pavankumar', 'password': '143Pavan..'})
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, 'registration/login.html')
+        self.assertEqual(response.context['form'].errors['username'], ['Please verify your email to login.'])
+
+    def test_to_activte_user(self):
+        """
```
Comment (bharathwaaj) [inline onboarding/tests/test_accounts/test_signup.py]: activate spelling

File: onboarding/tests/test_accounts/test_signup.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, 'registration/signup.html')
+
+    def test_post_signup_page(self):
+        """
+        Test to create user through signup page.
+        """
+        response = self.client.post(reverse('accounts:signup'), self.user)
+        self.assertEqual(response.context['domain'], 'testserver')
+        self.assertEqual(response.context['uid'], urlsafe_base64_encode(force_bytes(response.context['user'].pk)).decode())
```
Comment (bharathwaaj) [inline onboarding/tests/test_accounts/test_signup.py]: How is uid available in context ? It defeats the purpose of uid verification

File: onboarding/tests/test_accounts/test_signup.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+    def test_to_activte_user(self):
+        """
+        Test to get login page.
+        """
+        response = self.client.post(reverse('accounts:signup'), self.user)
+        self.assertEqual(response.status_code, 200)
+        user = User.objects.get(pk=response.context['user'].pk)
+        self.assertEqual(user.is_active, False)
+        activate = self.client.get(reverse('accounts:account_activate', kwargs={'uidb64': response.context['uid'], 'token': response.context['token']}))
```
Comment (bharathwaaj) [inline onboarding/tests/test_accounts/test_signup.py]: Same here

**Review by bharathwaaj — CHANGES_REQUESTED** [review]

File: onboarding/apps/accounts/views.py [hunk 7]
```diff
@@ -1,7 +1,59 @@
...
+        uid = urlsafe_base64_decode(uidb64).decode()
+        user = User.objects.get(pk=uid)
+    except(TypeError, ValueError, OverflowError, User.DoesNotExist):
+        user = None
+    if user is not None and account_activation_token.check_token(user, token):
+        user.is_active = True
+        user.save()
+        my_login(request, user)
+        return HttpResponse('<span>Thank you for your email confirmation. Now you can <a href="/login/">login</a> your account.</span>')
+    else:
```
[the remark this answers is not in the document]
Comment (pavan-testpress) [reply onboarding/apps/accounts/views.py]: If user tries to open the same link again it will display that else part message. don't we need that ?

**Pull request merged**